test/integration/targets/junos_l2_interface/tests/netconf/basic.yaml

```diff
@@ -0,0 +1,337 @@
+---
+- debug: msg="START junos_l2_interface netconf/basic.yaml on connection={{ ansible_connection }}"
+
+- name: setup - remove interface address
+  junos_l2_interface:
+    name: ge-0/0/1
+    state: absent
+    provider: "{{ netconf }}"
+
+- name: setup - remove interface address
+  junos_l2_interface:
+    name: ge-0/0/2
+    state: absent
+    provider: "{{ netconf }}"
+
+- name: Setup delete vlan 100
+  junos_vlan:
+    vlan_id: 100
+    name: red
+    state: absent
+    provider: "{{ netconf }}"
+  register: result
+
+- name: Setup delete vlan 200
+  junos_vlan:
+    vlan_id: 200
+    name: blue
+    state: absent
+    provider: "{{ netconf }}"
+  register: result
+
+- name: Setup delete vlan 300
+  junos_vlan:
+    vlan_id: 300
+    name: green
+    state: absent
+    provider: "{{ netconf }}"
+  register: result
+
+- name: Setup create vlan 100
+  junos_vlan:
+    vlan_id: 100
```

Comment:
I would probably `loop` these three if all you're changing is `vlan_id` and `name`